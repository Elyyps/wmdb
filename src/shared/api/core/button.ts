export interface IButton {
  fullWidth?: boolean;
  href?: string;
  icon?: string;
  onClick?: () => void;
  position?: string;
  target?: string;
  title: string;
  type?: string;
  variant?: string;
}
