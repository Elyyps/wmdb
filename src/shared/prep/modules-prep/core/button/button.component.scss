.button {
  margin: 0;
  border: solid #eaeaea;
  border-width: 1px 1px 3px;
  box-sizing: border-box;
  font-size: 16px;
  line-height: 1.3;
  font-weight: 600;
  font-family: $global-font-family;
  text-align: center;
  text-decoration: none;
  display: inline-flex;
  align-items: center;
  user-select: none;
  justify-content: center;
  position: relative;
  padding: 9px 26px;
  transition: background-color 0.3s;
  cursor: pointer;
  color: $global-color;
  border-radius: 3px;
  background-color: $bg-muted;
  @media (max-width: $breakpoint-small-max) {
    padding-left: 15px;
    padding-right: 15px;
  }
  @media (max-width: $breakpoint-xsmall-max) {
    width: 100%;
  }
  &.full{
    width: 100%;
  }
  &.large{
    padding: 14px 26px;
  }
  &:hover {
    text-decoration: none;
    background-color: #eaeaea;
    color: $global-color;
  }
  
  .icon-svg {
    line-height: .5;
    margin-left: 8px;
    path {
      fill: currentColor;
    }
  }

  &--primary {
    background: $brand-primary;
    color: $global-inverse-color;
    border-color: $brand-primary-dark;
    &:hover {
      background-color: $brand-primary-dark;
      color: $global-inverse-color;
    }
  }

  &--secondary {
    background: $brand-secondary;
    color: $global-inverse-color;
    border-color: $brand-secondary-dark;
    &:hover {
      background-color: $brand-secondary-dark;
      color: $global-inverse-color;
    }
  }

  
  &--icon-left {
    > .icon-svg {
      order: -1;
      margin-left: 0;
      margin-right: 8px;
    }
  }
}
