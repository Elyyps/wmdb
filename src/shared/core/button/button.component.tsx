import * as React from "react";
import styles from "./button-component.module.scss";
import { Link } from "react-router-dom";
import { IconComponent } from "@app/prep/modules-prep/core";
import ArrowDown from "@assets/icons/chevron-down.svg";
import classNames from "classnames";

export interface IButtonProps {
  children?: any;
  href?: string;
  icon?: any;
  iconStroke?: boolean;
  onClick?: any;
  position?: string;
  target?: string;
  title: string;
  type?: string;
  variant?: string;
}

const Button = (props: IButtonProps) => {
  const { onClick, href, variant, target, title, type, icon, position, iconStroke, ...other } = props;
  const classModify = variant || "default";
  const buttonClassName = classNames(styles["button"], {
    [styles[`button--${classModify}`]]: classModify,
    [styles[`button--icon-${position}`]]: position
  });

  return (
    <React.Fragment>
      {href ? (
        <Link {...other} to={href} className={buttonClassName} target={target}>
          <span>{title}</span>
          {icon && <IconComponent icon={icon} size="14px" stroke={iconStroke ? iconStroke : true} />}
        </Link>
      ) : (
        <button onClick={onClick} {...other} type={type} className={buttonClassName}>
          <span>{title}</span>
          {icon && <IconComponent icon={icon} size="14px" stroke={iconStroke ? iconStroke : true} />}
          {variant === "dropdown" && (
            <div className={styles["button__arrow"]}>
              <IconComponent icon={ArrowDown} size={"12px"} stroke={iconStroke ? iconStroke : true} />
            </div>
          )}
        </button>
      )}
    </React.Fragment>
  );
};

export { Button };
