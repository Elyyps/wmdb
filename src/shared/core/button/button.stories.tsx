import React from "react";
import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { Button } from "./button.component";
import Arrow from "@assets/icons/chevron-down.svg";

storiesOf("Button", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <div>
      <Button title="Primary" />
      <Button variant="secondary" title="Secondary" />
      <Button icon={Arrow} title="Primary with Icon" />
    </div>
  ));
