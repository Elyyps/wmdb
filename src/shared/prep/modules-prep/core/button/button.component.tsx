import * as React from "react";
import "./button.component.scss";
import { Link } from "react-router-dom";
import { IconComponent } from "@app/prep/modules-prep/core";
import ArrowDown from "@assets/icons/chevron-down.svg";
import classNames from "classnames";

export interface IButtonProps {
  children?: any;
  href?: string;
  icon?: any;
  onClick?: any;
  position?: string;
  target?: string;
  title: string;
  type?: string;
  variant?: string;
}

const Button = (props: IButtonProps) => {
  const { onClick, href, variant, target, title, type, icon, position, ...other } = props;
  const classModify = variant || "default";
  const buttonClassName = classNames("button", {
    [`button--${classModify}`]: classModify,
    [`button--icon-${position}`]: position
  });

  return (
    <React.Fragment>
      {href ? (
        <Link {...other} to={href} className={buttonClassName} target={target}>
          <span>{title}</span>
          {icon ? <IconComponent icon={icon} size="14px" /> : ""}
        </Link>
      ) : (
        <button onClick={onClick} {...other} type={type} className={buttonClassName}>
          <span>{title}</span>
          {icon ? <IconComponent icon={icon} size="14px" /> : ""}
          {variant === "dropdown" && (
            <div className="button__arrow">
              <IconComponent icon={ArrowDown} size={"12px"} />
            </div>
          )}
        </button>
      )}
    </React.Fragment>
  );
};

export { Button };
