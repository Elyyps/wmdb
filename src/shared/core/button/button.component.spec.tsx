import React from "react";
import { shallow } from "enzyme";
import { Button } from "./button.component";
import Arrow from "@assets/icons/chevron-down.svg";

describe("[Button]", () => {
  it("should render without crashing", () => {
    const renderedComponent = shallow(<Button title="Primary" />);
    expect(renderedComponent).toMatchSnapshot();
  });
  it("should render without crashing", () => {
    const renderedComponent = shallow(<Button variant="secondary" title="Secondary" />);
    expect(renderedComponent).toMatchSnapshot();
  });
  it("should render without crashing", () => {
    const renderedComponent = shallow(<Button icon={Arrow} title="Primary" />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
