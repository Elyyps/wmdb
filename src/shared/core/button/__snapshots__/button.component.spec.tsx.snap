// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Button] should render without crashing 1`] = `
<Fragment>
  <button
    className="button button--default"
  >
    <span>
      Primary
    </span>
  </button>
</Fragment>
`;

exports[`[Button] should render without crashing 2`] = `
<Fragment>
  <button
    className="button button--secondary"
  >
    <span>
      Secondary
    </span>
  </button>
</Fragment>
`;

exports[`[Button] should render without crashing 3`] = `
<Fragment>
  <button
    className="button button--default"
  >
    <span>
      Primary
    </span>
    <IconComponent
      icon="chevron-down.svg"
      size="14px"
      stroke={true}
    />
  </button>
</Fragment>
`;
